src/managers/SessionManager.ts

```diff
@@ -78,15 +78,23 @@ class SessionManager {
             sessionId: sessionId,
             playerName: PlayerManager.generateName(),
         } as JoinDTO;
-        const response: axios.AxiosResponse<UUID> = await api.post(
-            "/join",
-            requestBody,
-        );
-        if (!response.data) {
+        try {
+            const response: axios.AxiosResponse<UUID> = await api.post(
+                "/join",
+                requestBody,
+            );
+            PlayerManager.saveId(response.data);
+            location.pathname = "";
+        } catch (error) {
+            if (axios.isAxiosError(error) && error.response?.status) {
+                alert(error.response?.data.message);
+                PlayerManager.delete();
+                location.pathname = "";
+                return;
+            }
+            alert("Unexpected error when joining game. Please try again.");
             return;
         }
-        PlayerManager.saveId(response.data);
-        location.pathname = "";
```

Comment:
i think the location line could stay here either way and remove it from the new lines above

Reply:
Good point